libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookup.java

```diff
@@ -10,14 +10,29 @@
 package org.elasticsearch.entitlement.runtime.policy;
 
 import java.nio.file.Path;
-import java.util.function.Function;
 import java.util.stream.Stream;
 
-public record PathLookup(
-    Path homeDir,
-    Path configDir,
-    Path[] dataDirs,
-    Path[] sharedRepoDirs,
-    Path tempDir,
-    Function<String, Stream<String>> settingResolver
-) {}
+/**
+ * Resolves paths for known directories checked by entitlements.
+ */
+public interface PathLookup {
+    enum BaseDir {
+        HOME,
```

Comment:
nit: I think this could be confused with Elasticsearch home directory, can we call this `USER_HOME`?

Reply:
Changed.